src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -879,6 +961,43 @@ public unsafe void CopyTo(Array array, int index)
                         }
                     }
                 }
+                else if (AdvSimd.IsSupported)
+                {
+                    Vector128<byte> bitMask = Vector128.Create(0x80402010_08040201).AsByte();
+                    Vector128<byte> ones = Vector128.Create((byte)1);
+                    fixed (bool* destination = &boolArray[index])
+                    {
+                        for (; (i + Vector128<byte>.Count * 2) <= m_length; i += Vector128<byte>.Count * 2)
+                        {
+                            int bits = m_array[i / BitsPerInt32];
+                            // Same logic as SSSE3 path, except we do not have Shuffle instruction.
+                            // (TableVectorLookup could be an alternative - dotnet/runtime#1277)
+                            // Instead we use chained ZIP1/2 instructions:
+                            // (A0 is the byte containing LSB, A3 is the byte containing MSB)
+                            // bits                             - A3 A2 A1 A0
+                            // Byte reversal                    - A0 A1 A2 A3
+                            // v1 = Vector128.Create            - A0 A1 A2 A3 A0 A1 A2 A3 A0 A1 A2 A3 A0 A1 A2 A3
+                            // v2 = ZipLow(v1, v1)              - A0 A0 A1 A1 A2 A2 A3 A3 A0 A0 A1 A1 A2 A2 A3 A3
+                            // v3 = ZipLow(v2, v2)              - A0 A0 A0 A0 A1 A1 A1 A1 A2 A2 A2 A2 A3 A3 A3 A3
+                            // shuffledLower = ZipLow(v3, v3)   - A0 A0 A0 A0 A0 A0 A0 A0 A1 A1 A1 A1 A1 A1 A1 A1
+                            // shuffledHigher = ZipHigh(v3, v3) - A2 A2 A2 A2 A2 A2 A2 A2 A3 A3 A3 A3 A3 A3 A3 A3
+
+                            Vector128<byte> vector = Vector128.Create(BinaryPrimitives.ReverseEndianness(bits)).AsByte();
```

Comment:
Is `BinaryPrimitives.ReverseEndianness` optimized to `bswap` on AArch64 ?

Reply:
I do not think so, which is why I left a comment on #33308 about it. It would be `REV` instructions that correspond to x86's bswap instructions.